tuf/ngclient/updater.py

```diff
@@ -262,10 +339,16 @@ def _download_metadata(
     ) -> bytes:
         """Download a metadata file and return it as bytes"""
         encoded_name = parse.quote(rolename, "")
-        if version is None:
-            url = f"{self._metadata_base_url}{encoded_name}.json"
+
+        if self._spec_version_dir == "":
```

Comment:
IMO this is getting really hard to read. I think it would benefit a lot from a:

```python
def _format_url(self, base: str, parts: List[str]):
    return f"{base.strip('/')}/{'/'.join([p.strip('/') for p in parts])"

def _format_metadata_basename(rolename: str, version: Optional[int]) -> str:
    encoded_name = parse.quote(rolename, "")
    if version is None:
        return f"{encoded_name.json}"
    return f"{version}.{encoded_name}.json"
```

So you could build the URL up as:

```python
url = self._format_url(filter(None, [self._spec_version_dir, _format_basename(encoded_name, version)]))
```